src/main/scala/arcana/world/ArgumentEvaluator.scala

```diff
@@ -0,0 +1,41 @@
+package arcana.world
+
+import scala.util.Try
+
+class ArgumentEvaluator {
+  def evaluat(args: Array[String]): Unit = {
+    import ArgumentEvaluator._
+
+    var needTypeInfo: Boolean = false // TODO add more judge flag
```

Comment:
You may need another options class for **all** terminal arguments. e.g.
```scala
case class Arguments(val needTypeInfo: Boolean = false)
```
Then you can use `fold` instead of a mutable variable.